channels/api.py

```diff
@@ -434,9 +434,20 @@ def add_moderator(self, moderator_name, channel_name):
             raise NotFound("User {} does not exist".format(moderator_name))
 
         self.get_channel(channel_name).moderator.add(user)
-
+        new_moderator = Api(user)
+        new_moderator.accept_invite(channel_name)
+        del new_moderator
```

Comment:
FYI You generally should never need to do this, it will get deleted when the local variable goes out of scope (when the reference count is 0)